apps/passport-server/src/database/queries/pretixEventInfo.ts

```diff
@@ -30,6 +30,7 @@ export async function insertPretixEventsInfo(
     `\
       insert into devconnect_pretix_events_info (event_name, pretix_events_config_id, checkin_list_id, is_deleted)
       values ($1, $2, $3, FALSE)
+      on conflict (pretix_events_config_id) do update set event_name = $1, checkin_list_id = $3
```

Comment:
overall I'm pretty sensitive to changes to these files. I wonder what is motivating you to make this change? to the extent that is possible, I'd like changes to the database functions to be tested at least in a cursory manner.

Reply:
I added all the `on conflict` clauses so that the scratch dev CLI could bootstrap local testing events and tickets idempotently so one could safely rerun them any number of time and resume if something broke in mid-way.

+1 we should add testing for these and double check if `do update` clauses are safe.

Reply:
In general, the `pretix_*` tables are managed manually, but the `devconnect_*` tables are managed by the sync service. I _think_ these changes are fine, but I am assuming that if you're populating these tables manually it's because you have the sync service turned off (it's off by default since a change to environment variables a week or two ago).

Reply:
@robknight I confirmed that we have `PRETIX_SYNC_DISABLED=TRUE` in local `.env`. One suggestion to address https://github.com/proofcarryingdata/zupass/pull/624#discussion_r1329471466 is that we add a new flag or use `NODE_ENV` for a `development` mode. If `NODE_ENV != development`,  manually writing to the `pretix_*` tables could be disabled.